pkg/authorization/api/types.go

```diff
@@ -75,6 +75,8 @@ type PolicyRule struct {
 	AttributeRestrictions kruntime.EmbeddedObject `json:"attributeRestrictions"`
 	// Resources is a list of resources this rule applies to.  ResourceAll represents all resources.
 	Resources []string `json:"resources"`
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	ResourceNames kutil.StringSet
```

Comment:
> json tag, probably with omitempty?

You suggested leaving the off the in-memory types and same reason as above for not stripping all of them.